utils/webScraper.js

```diff
@@ -1,89 +1,85 @@
 // const travel = 'travel-and-outdoor';
 const locationSample = 'port-harcourt';
 // variable samples
+const fetchData = async () => {
+  const result = await axios({
+    method: "get",
+    url: `https://www.eventbrite.com/d/nigeria--${locationSample}/${science}--events/`,
+    json: true,
+    headers: { 'User-Agent': 'Mozilla/5.0' },
+  });
+  return cheerio.load(result.data, {
+    normalizeWhitespace: false,
+  });
+};
 
-export default function ScrapeEvents() {
-  const fetchData = async () => {
-    const result = await axios({
-      method: "get",
-      url: `https://www.eventbrite.com/d/nigeria--${locationSample}/${science}--events/`,
-      json: true,
-      headers: { 'User-Agent': 'Mozilla/5.0' },
-    });
-    return cheerio.load(result.data, {
-      normalizeWhitespace: false,
-    });
-  };
-
-  fetchData().then(($) => {
+const ScrapeEvents = new Promise((resolve, reject) => {
+  return fetchData().then(($) => {
     const datesArray = [];
     const titlesArray = [];
     const linksArray = [];
     const imagesArray = [];
     const locationsArray = [];
     const pricesArray = [];
     const dataSet = [];
-
     const dates = $('.eds-text-color--primary-brand').toArray();
     const titles = $('.eds-media-card-content__action-link').toArray();
     const links = $('.eds-media-card-content__action-link').toArray();
     const images = $('.eds-max-img').toArray();
     const locations = $('.card-text--truncated__one').toArray();
     const prices = $('.eds-media-card-content__flag').toArray();
-    const eventLinksArray = []; 
-
+    const eventLinksArray = [];
     images.forEach((image) => {
       const imageLink = $(image).attr('src');
       imagesArray.push(imageLink);
     });
-
     dates.forEach((item) => {
       datesArray.push($(item).text());
     });
-
     titles.forEach((title) => {
       titlesArray.push($(title).text());
     });
-
     locations.forEach((location) => {
       locationsArray.push($(location).text());
     });
-
     prices.forEach((price) => {
       pricesArray.push($(price).text());
     });
-
     links.forEach((link) => {
       const fullLink = $(link).attr('href');
       const start = fullLink.indexOf('ts-') + 3;
       const end = fullLink.indexOf('?') - fullLink.indexOf('ts-') - 3;
       linksArray.push(fullLink.substr(start, end));
       eventLinksArray.push(fullLink);
     });
-
     const newTitles = titlesArray.filter((str) => /\S/.test(str));
-
-    for (let i = 0; i < datesArray.length; i++) {
+    const uniqueTitles = Array.from(new Set(newTitles));
+    const uniqueDates = Array.from(new Set(datesArray));
+    const uniqueLinkIds = Array.from(new Set(linksArray));
+    const uniqueImageLinks = Array.from(new Set(imagesArray));
+    const uniqueLocations = Array.from(new Set(locationsArray));
+    const uniquePrices = Array.from(new Set(pricesArray));
+    const uniqueEventLinks = Array.from(new Set(eventLinksArray));
+    for (let i = 0; i < uniqueDates.length; i++) {
```

Comment:
This could be cleaner and D>R>Y than this

Reply:
We can work on this in the next iteration or maybe tomorrow when we attempt the dynamic scraper.